libfwupdplugin/fu-sbatlevel-section.c

```diff
@@ -26,22 +28,21 @@ fu_sbatlevel_section_add_entry(FuFirmware *firmware,
 			       FwupdInstallFlags flags,
 			       GError **error)
 {
-	gsize bufsz = 0;
-	const guint8 *buf;
-	gsize size = 0;
+	gsize streamsz = 0;
 	g_autoptr(FuFirmware) entry_fw = NULL;
-	g_autoptr(GBytes) entry_blob = NULL;
-	g_autoptr(GBytes) fw = NULL;
+	g_autoptr(GInputStream) partial_stream = NULL;
 
-	fw = fu_input_stream_read_bytes(stream, offset, G_MAXSIZE, error);
-	if (fw == NULL)
+	/* stop at the null terminator */
+	if (!fu_input_stream_size(stream, &streamsz, error))
 		return FALSE;
-	buf = g_bytes_get_data(fw, &bufsz);
-
-	/* look for the null terminator */
-	for (size = 0; ((offset + size) < bufsz); ++size) {
-		if (buf[offset + size] == 0)
+	for (guint i = offset; i < streamsz; i++) {
+		guint8 value = 0;
+		if (!fu_input_stream_read_u8(stream, i, &value, error))
+			return FALSE;
+		if (value == 0x0) {
+			streamsz = i - 1;
```

Comment:
I suppose we used to parse with the nul character included but now it does not take it into account. I suppose this is fine.

Reply:
I don't think it matters; I think it's probably preferable to pass in the size without the NULL to just match something like strlen().